src/main/java/seedu/address/logic/commands/CategorizeTagCommand.java

```diff
@@ -26,32 +27,35 @@ public class CategorizeTagCommand extends Command {
     public static final String MESSAGE_TAG_NOT_EXIST = "Tag not found: %1$s";
     public static final String MESSAGE_INVALID_CATEGORY = "Invalid category: %1$s";
     public static final String MESSAGE_DUPLICATE_CATEGORY = "Current category of %s is already %s";
-
-    private final Tag targetTag;
+    private final List<Tag> targetTags;
     private final TagCategory updatedCategory;
 
     /**
      * Constructs a command to categorize a tag.
      */
-    public CategorizeTagCommand(Tag targetTag, TagCategory updatedCategory) {
-        requireNonNull(targetTag);
+    public CategorizeTagCommand(List<Tag> targetTags, TagCategory updatedCategory) {
+        requireNonNull(targetTags);
         requireNonNull(updatedCategory);
-        this.targetTag = targetTag;
+        this.targetTags = targetTags;
         this.updatedCategory = updatedCategory;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         assert model != null;
-        if (!model.containsTag(targetTag)) {
-            throw new CommandException(String.format(MESSAGE_TAG_NOT_EXIST, targetTag));
-        }
-        TagCategory existingCat = model.getTagCategory(targetTag);
-        if (isDuplicateCategory(existingCat)) {
-            throw new CommandException(String.format(MESSAGE_DUPLICATE_CATEGORY, targetTag, updatedCategory));
```

Comment:
I think we still need to show an error and block the command when the category is a duplicate. This is to prevent a unchanged state from being saved to the history stack.